src/common/lib/python/common_extension.cc

```diff
@@ -452,12 +452,16 @@ PyTypeObject PyTaskType = {
 };
 
 /* Create a PyTask from a C struct. The resulting PyTask takes ownership of the
- * TaskSpec and will deallocate the TaskSpec in the PyTask destructor. */
-PyObject *PyTask_make(TaskSpec *task_spec, int64_t task_size) {
+ * task reply and will deallocate the task reply in the PyTask destructor. */
+PyObject *PyTask_make(uint8_t *task_reply, int64_t task_reply_size) {
```

Comment:
So, the local scheduler sends a `task_reply` object to the client. The `task_reply` contains a `task_spec`. So when we're done using the `task_spec`, we need to free the `task_reply`. I suppose we could avoid this just by copying the `task_spec`, which may be the way to go.